pkg/xds/bootstrap/testdata/bootstrap.gateway.golden.yaml

```diff
@@ -134,8 +134,13 @@ staticResources:
   secrets:
   - name: cp_validation_ctx
     validationContext:
-      matchSubjectAltNames:
-      - exact: localhost
+      matchTypedSubjectAltNames:
+      - matcher:
+          exact: localhost
+        sanType: DNS
+      - matcher:
+          exact: localhost
+        sanType: IP_ADDRESS
```

Comment:
I'm a bit uncomfortable about this. I think we should only use `IP_ADDRESS` if we know there is IP Address there.
I'd either try to infer if we have IP or DNS and set proper sanType or go with `SAN_TYPE_UNSPECIFIED`

Reply:
San type unspecified seems to be an invalid value. The behavior of the deprecated option is to create a matcher _for all 4_ types so this is already trimming down the matchers

Reply:
Ok

Reply:
Just for posterity https://github.com/envoyproxy/envoy/blob/ef08b1c3d0cf9c2af84d32529a11b3e2056e9dcb/source/common/ssl/certificate_validation_context_config_impl.cc#L79